pkg/storage/business_case.go

```diff
@@ -200,3 +202,110 @@ func (s *Store) CreateBusinessCase(businessCase *models.BusinessCase) (*models.B
 
 	return businessCase, nil
 }
+
+// UpdateBusinessCase creates a business case
+func (s *Store) UpdateBusinessCase(businessCase *models.BusinessCase) (*models.BusinessCase, error) {
+	// We are explicitly not updating ID, EUAUserID and SystemIntakeID
+	const updateBusinessCaseSQL = `
+		UPDATE business_case 
+		SET
+			project_name = :project_name,
+			requester = :requester,
+			requester_phone_number = :requester_phone_number,
+			business_owner = :business_owner,
+			business_need = :business_need,
+			cms_benefit = :cms_benefit,
+			priority_alignment = :priority_alignment,
+			success_indicators = :success_indicators,
+			as_is_title = :as_is_title,
+			as_is_summary = :as_is_summary,
+			as_is_pros = :as_is_pros,
+			as_is_cons = :as_is_cons,
+			as_is_cost_savings = :as_is_cost_savings,
+			preferred_title = :preferred_title,
+			preferred_summary = :preferred_summary,
+			preferred_acquisition_approach = :preferred_acquisition_approach,
+			preferred_pros = :preferred_pros,
+			preferred_cons = :preferred_cons,
+			preferred_cost_savings = :preferred_cost_savings,
+			alternative_a_title = :alternative_a_title,
+			alternative_a_summary = :alternative_a_summary,
+			alternative_a_acquisition_approach = :alternative_a_acquisition_approach,
+			alternative_a_pros = :alternative_a_pros,
+			alternative_a_cons = :alternative_a_cons,
+			alternative_a_cost_savings = :alternative_a_cost_savings,
+			alternative_b_title = :alternative_b_title,
+			alternative_b_summary = :alternative_b_summary,
+			alternative_b_acquisition_approach = :alternative_b_acquisition_approach,
+			alternative_b_pros = :alternative_b_pros,
+			alternative_b_cons = :alternative_b_cons,
+			alternative_b_cost_savings = :alternative_b_cost_savings,
+			updated_at = :updated_at
+		WHERE business_case.id = :id
+	`
+	const deleteLifecycleCostsSQL = `
+		DELETE FROM estimated_lifecycle_cost
+		WHERE business_case = :id
+	`
+
+	tx := s.DB.MustBegin()
+	//Rollback only happens if transaction isn't committed
+	defer tx.Rollback()
+	result, err := tx.NamedExec(updateBusinessCaseSQL, &businessCase)
+	if err != nil {
+		s.logger.Error(
+			fmt.Sprintf("Failed to update business case %s", err),
+			zap.String("id", businessCase.ID.String()),
+		)
+		return businessCase, err
+	}
+	affectedRows, rowsAffectedErr := result.RowsAffected()
+	if affectedRows == 0 || rowsAffectedErr != nil {
+		s.logger.Error(
+			fmt.Sprintf("Failed to update business case %s", err),
+			zap.String("id", businessCase.ID.String()),
+		)
+		return businessCase, errors.New("business case not found")
+	}
+
+	_, err = tx.NamedExec(deleteLifecycleCostsSQL, &businessCase)
+	if err != nil {
+		s.logger.Error(
+			fmt.Sprintf("Failed to update pre-existing business case costs %s", err),
+			zap.String("id", businessCase.ID.String()),
+		)
+		return businessCase, err
+	}
+
+	for _, cost := range businessCase.LifecycleCostLines {
```

Comment:
I think this chunk of adding a new lifecycle Cost item can be factored out of this and the POST storage method.

Reply:
nice call. Will move it out!